### PR TITLE
docs: add code-titles and write some docs

### PR DESCRIPTION
**Pull request by juliusmarminge**
Closes #<issue>

## ✅ Checklist

- [ ] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [ ] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] I performed a functional test on my final commit

---

## Changelog

_[Short description of what has changed]_

---

## Screenshots

_[Screenshots]_

💯

### CODE REVIEW

File: www/src/pages/en/t3-collection.md [hunk 1]
```diff
@@ -8,14 +8,14 @@ Made a project using the T3 stack and want to share it? Add it to the list!
...
-| ----------------------------------------------------------------- | --------------------------------------------------------------------- | --------------------------------------------------------------------- |
-| Zapdos - a QnA app for streamers                                  | [t3-oss/zapdos](https://github.com/t3-oss/zapdos)                     | [z.t3.gg](https://z.t3.gg)                                            |
-| Shoutify - Free, open-source, self-hosted social media management | [techsquidtv/shoutify](https://github.com/TechSquidTV/Shoutify)       | [shoutify.app](https://github.com/TechSquidTV/Shoutify) (coming soon) |
-| nexxel's personal site                                            | [nexxeln/nexxel.dev](https://github.com/nexxeln/nexxel.dev)           | [nexxel.dev](https://nexxel.dev)                                      |
-| Julius's personal site                                            | [juliusmarminge/jumr.dev](https://github.com/juliusmarminge/jumr.dev) | [jumr.dev](https://jumr.dev)                                          |
```
Comment (juliusmarminge) [inline www/src/pages/en/t3-collection.md:16]: removed mine cause its no longer using fancy t3-tech

File: www/src/pages/en/usage/next-auth.md [hunk 2]
```diff
@@ -56,11 +65,65 @@ declare module "next-auth" {
...
+```
+
+Located at `server/router/context.ts`, the context provider is setup to recieve the `req` and `res` object from NextJS, to query if a current session exists and provide it to the tRPC context. This allows you to use the `session` object in your API routes to check if a user is authenticated in middleware.
+
 ## Usage with Prisma
 
 TODO: what we setup, how to add more fields to user and acc tables.
-
-## Middleware
+FIXME: FIXME: FIXME:
```
Comment (juliusmarminge) [inline www/src/pages/en/usage/next-auth.md:126]: feel free to push text directly to this branch here

File: www/src/styles/global.css [hunk 3]
```diff
@@ -153,8 +153,13 @@
           hover:text-t3-purple-700 dark:text-t3-purple-200 
           dark:decoration-t3-purple-200 dark:hover:text-t3-purple-300;
   }
+
+  .markdown .remark-code-title {
+    @apply z-10 -mb-1 mt-3 w-full rounded-t-md border-b-2 border-t3-purple-200 bg-[#292D3E] px-2 py-1 text-t3-purple-200;
+  }
```
Comment (juliusmarminge) [inline www/src/styles/global.css:159]: feel free to improve the styling of the titles

File: www/src/pages/en/usage/next-auth.md [hunk 4]
```diff
@@ -56,11 +65,76 @@ declare module "next-auth" {
...
 
+The same pattern can be used to add any other data to the `session` object, such as a `role` field, but **should not be misused to store sensitive data** on the client.
+
+### Usage with tRPC
+
+When using NextAuth.js with tRPC, you can create reusable, protected procedures using [middleware](https://trpc.io/docs/v10/middlewares). This allows you to create procedures that can only be accessed by authenticated users. `create-t3-app` sets all of this up for you, allowing you to easily access the session object within authenticated procedures.
+
+This is done in a two step process:
+
+1. Grab the session from the request headers using the [unstable_getServerSession](https://next-auth.js.org/configuration/nextjs#unstable_getserversession) function (don't worry, its not unstable cause of security reasons, but cause of its API might change). The advantage of using `unstable_getServerSession` instead of the regular `getSession` is that its a server-side only function and doesn't trigger unnecessary fetch calls. `create-t3-app` creates a helper function that abstacts this weird, funky API away.
```
Comment (juliusmarminge) [inline www/src/pages/en/usage/next-auth.md:76]: "that abstacts this weird, funky API away."

it's late. words are hard

Comment (nexxeln) [reply www/src/pages/en/usage/next-auth.md:76]: maybe peculiar

File: www/src/pages/en/usage/trpc.md [hunk 5]
```diff
@@ -118,12 +116,51 @@ const UserPage = () => {
...
+    res.status(500).json({ message: "Internal server error" });
+  }
+};
+
+export default userByIdHandler;
+```
+
+### Exposing every procedure as a REST endpoint
+
+If you want to expose every single procedure externally, checkout the community build plugin [trpc-openapi](https://github.com/jlalmes/trpc-openapi/tree/master). By providing some extra meta-data to your procedures, you can generate a OpenAPI compliant REST API from your tRPC router.
```
Comment (asrvd) [inline www/src/pages/en/usage/trpc.md:157]:
```suggestion
If you want to expose every single procedure externally, checkout the community built plugin [trpc-openapi](https://github.com/jlalmes/trpc-openapi/tree/master). By providing some extra meta-data to your procedures, you can generate an OpenAPI compliant REST API from your tRPC router.
```

**Pull request merged**